src/api/index.ts

```diff
@@ -0,0 +1,118 @@
+import { RULE } from "../constants/rules";
+import {
+  CART_ITEMS_COUNTS_ENDPOINT,
+  CART_ITEMS_ENDPOINT,
+  PRODUCTS_ENDPOINT,
+} from "./endpoints";
+import { fetchWithAuth } from "./fetchWithAuth";
+
+interface QueryParams {
+  [key: string]:
+    | undefined
+    | string
+    | number
+    | boolean
+    | (string | number | boolean)[];
+}
+
+export interface GetProductsParams {
+  category?: Category;
+  page?: number;
+  size?: number;
+  sort?: Sort;
+}
+
+const createQueryString = (params: QueryParams): string => {
+  return Object.entries(params)
+    .map(([key, value]) => {
+      if (value === undefined) {
+        return;
+      }
+      if (Array.isArray(value)) {
+        return `${encodeURIComponent(key)}=${encodeURIComponent(
+          value.join(",")
+        )}`;
+      }
+      return `${encodeURIComponent(key)}=${encodeURIComponent(String(value))}`;
+    })
+    .join("&");
+};
```

Comment:
하나의 `map` 메서드에서 모든 동작을 수행하는 대신 method chaining을 이용해서 조금 더 읽기 쉽게 만들어 볼 수 있을거 같아요
```suggestion
const createQueryString = (params) => {
  return Object.entries(params)
    .filter(([key, value]) => value !== undefined)
    .map(([key, value]) => {
      if (Array.isArray(value)) {
        return value.map(v => `${encodeURIComponent(key)}=${encodeURIComponent(v)}`).join('&');
      }
      return `${encodeURIComponent(key)}=${encodeURIComponent(String(value))}`;
    })
    .join('&');
};
```

Reply:
확실히 제안 해 주신 방법이 가독성도 올라가고 훨씬 개발자스럽(?)네요! 반영하겠습니다 :)